In [1]:
import pandas as pd

In [2]:
movie_column_names = ["WikiID", "FreeID", "Title", "RelDate", "Revenue", "Runtime", "Languages", "Countries", "Genres"]
character_column_names = ["WikiID", "FreeID", "MovieRelDate", "CharName", "DOB", "Gender", "Height", "Ethnicity", "Actor", "Age", "FreeMapID", "FreeCharID", "FreeActorID"]
movies_path = "movie.metadata.tsv"
character_path = "character.metadata.tsv"
movies = pd.read_csv(movies_path, sep='\t', header=None, names=movie_column_names)
characters = pd.read_csv(character_path, sep='\t', header=None, names=character_column_names)

In [20]:
display(movies.head())
display(characters.head())

WikiID      FreeID                                              Title  \
0    975900   /m/03vyhn                                     Ghosts of Mars   
1   3196793   /m/08yl5d  Getting Away with Murder: The JonBenét Ramsey ...   
2  28463795  /m/0crgdbh                                        Brun bitter   
3   9363483  /m/0285_cd                                   White Of The Eye   
4    261236   /m/01mrr1                                  A Woman in Flames   

      RelDate     Revenue  Runtime                           Languages  \
0  2001-08-24  14010832.0     98.0  {"/m/02h40lc": "English Language"}   
1  2000-02-16         NaN     95.0  {"/m/02h40lc": "English Language"}   
2        1988         NaN     83.0  {"/m/05f_3": "Norwegian Language"}   
3        1987         NaN    110.0  {"/m/02h40lc": "English Language"}   
4        1983         NaN    106.0   {"/m/04306rv": "German Language"}   

                                   Countries  \
0  {"/m/09c7w0": "United States of America"}   
1  {"/m/09c7w0": "United States of America"}   
2                     {"/m/05b4w": "Norway"}   
3             {"/m/07ssc": "United Kingdom"}   
4                    {"/m/0345h": "Germany"}   

                                              Genres  
0  {"/m/01jfsb": "Thriller", "/m/06n90": "Science...  
1  {"/m/02n4kr": "Mystery", "/m/03bxz7": "Biograp...  
2  {"/m/0lsxr": "Crime Fiction", "/m/07s9rl0": "D...  
3  {"/m/01jfsb": "Thriller", "/m/0glj9q": "Erotic...  
4                            {"/m/07s9rl0": "Drama"}

WikiID     FreeID MovieRelDate                    CharName         DOB  \
0  975900  /m/03vyhn   2001-08-24                    Akooshay  1958-08-26   
1  975900  /m/03vyhn   2001-08-24  Lieutenant Melanie Ballard  1974-08-15   
2  975900  /m/03vyhn   2001-08-24         Desolation Williams  1969-06-15   
3  975900  /m/03vyhn   2001-08-24          Sgt Jericho Butler  1967-09-12   
4  975900  /m/03vyhn   2001-08-24             Bashira Kincaid  1977-09-25   

  Gender  Height   Ethnicity               Actor   Age   FreeMapID  \
0      F   1.620         NaN      Wanda De Jesus  42.0  /m/0bgchxw   
1      F   1.780  /m/044038p  Natasha Henstridge  27.0   /m/0jys3m   
2      M   1.727     /m/0x67            Ice Cube  32.0   /m/0jys3g   
3      M   1.750         NaN       Jason Statham  33.0  /m/02vchl6   
4      F   1.650         NaN         Clea DuVall  23.0  /m/02vbb3r   

   FreeCharID FreeActorID  
0  /m/0bgcj3x  /m/03wcfv7  
1  /m/0bgchn4   /m/0346l4  
2  /m/0bgchn_  /m/01vw26l  
3  /m/0bgchnq   /m/034hyc  
4  /m/0bgchp9   /m/01y9xg

In [25]:
n_mov = len(movies)
n_char = len(characters)
print('Number of rows in the movie.metadata dataset :', n_mov)
print('Number of rows in the movie.metadata dataset :', n_char)

Number of rows in the movie.metadata dataset : 81741
Number of rows in the movie.metadata dataset : 228141


### Check for missing entries :
In which columns are they present, and for those columns : what is the percentage how missing entries.

In [4]:
movies.isna().any()

WikiID       False
FreeID       False
Title        False
RelDate       True
Revenue       True
Runtime       True
Languages    False
Countries    False
Genres       False
dtype: bool

In [31]:
print('Percentage of missing entries in the movie dataset:\n', 100*movies[['RelDate', 'Revenue', 'Runtime']].isna().sum() / n_mov)

Percentage of missing entries in the movie dataset:
 RelDate     8.443743
Revenue    89.722416
Runtime    25.018045
dtype: float64


There are 90% of revenues that are non specified. We probably won't be able to use this feature.  
8% of the release dates and 25% of the runtimes are missing, we can fill them if we find the correct ones.

In [5]:
characters.isna().any()

WikiID          False
FreeID           True
MovieRelDate     True
CharName         True
DOB              True
Gender           True
Height           True
Ethnicity        True
Actor            True
Age              True
FreeMapID        True
FreeCharID       True
FreeActorID      True
dtype: bool

In [35]:
print('Percentage of missing entries in the character dataset:\n', 100*characters[['FreeID', 'MovieRelDate', 'CharName',
                                                                                   'DOB', 'Gender', 'Height', 'Ethnicity',
                                                                                   'Actor', 'Age', 'FreeMapID', 'FreeCharID',
                                                                                   'FreeActorID']].isna().sum() / n_char)

Percentage of missing entries in the character dataset:
 FreeID           0.000438
MovieRelDate     2.279731
CharName        57.650313
DOB             23.462683
Gender          10.277854
Height          65.654573
Ethnicity       76.448775
Actor            0.281405
Age             35.129153
FreeMapID        0.000438
FreeCharID      57.649874
FreeActorID      0.183658
dtype: float64


65% of the actors height information is missing, and 76% of the actors ethnicities are missing.  
For the characters, 57% of the names are missing.